lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -49,6 +49,12 @@ def as_test_response(obj)
               resp.request = ActionDispatch::Request.new({})
             end
             ::ActionDispatch::TestResponse.from_response(obj)
+          elsif obj.respond_to?(:status) && obj.respond_to?(:headers) && obj.respond_to?(:body)
```

Comment:
rack-test is a [runtime dependency](https://rubygems.org/gems/actionpack) of actionpack. How comes it won't be loaded?

Reply:
PS which is weird, because rack-test will also load in the production env.

Reply:
I think the comment is copy-pasta from the other implementation

Reply:
I assumed that `Rack::MockResponse` was not available, but I did not check.

Now that I have checked, it is provided by [rack](https://github.com/rack/rack/blob/main/lib/rack/mock_response.rb) itself, so I can simplify the condition.